py-polars/polars/series/struct.py

```diff
@@ -73,24 +73,25 @@ def field(self, name: str) -> Series:
         ]
         """
 
-    def rename_fields(self, names: Sequence[str]) -> Series:
+    def rename_fields(self, names: Sequence[str], strict: bool = True) -> Series:
         """
         Rename the fields of the struct.
 
         Parameters
         ----------
         names
             New names in the order of the struct's fields.
+        strict
+            If True, raises an error if the length of names does not match the number of fields.
 
         Examples
         --------
         >>> s = pl.Series([{"a": 1, "b": 2}, {"a": 3, "b": 4}])
-        >>> s.struct.fields
-        ['a', 'b']
-        >>> s = s.struct.rename_fields(["c", "d"])
-        >>> s.struct.fields
-        ['c', 'd']
+        >>> s.struct.rename_fields(["c", "d"], strict=True)
```

Comment:
I reckon the example would be more informative if you:
- show the error you get if you try to rename but provide the wrong number of names with `strict=True`;
- show what happens if you try to rename but provide the wrong number of names with `strict=False`;
- both of the above